pkg/metrics/metrics.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2018 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// ServerMetrics Prometheus metrics for the server
+type ServerMetrics struct {
+	/****Metrics related to Ark server backup***********/
+	lastBackupStatus prometheus.Gauge
+}
+
+const (
+	lastBackupStatus = "last_backup_status"
+)
+
+// NewServerMetrics returns new ServerMetrics
+func NewServerMetrics(namespace string) *ServerMetrics {
+	return &ServerMetrics{
+		lastBackupStatus: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: namespace,
+				Name:      lastBackupStatus,
+				Help:      "Status of the most recent backup. 1 if it was successful, 0 otherwise",
+			},
+		),
+	}
+}
+
+// GetMetricsMux returns a mux with the metrics handle
```

Comment:
Let's get rid of this function and set up the mux where you spawn the goroutine in server.go so this package doesn't need to know anything about http.